screens/AuthScreen/AuthScreen.js

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable max-statements */
+import React, { useState, useEffect } from 'react';
+import { View, ScrollView } from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import { Input, Button, Divider, Text } from 'react-native-elements';
+import { useDispatch, useSelector } from 'react-redux';
+import { LOGIN_REQUEST, REGISTER_REQUEST } from '../../store/types';
+import Colors from '../../constants/Colors';
+import { styles } from '../../components/styles';
+
+function AuthScreen(props) {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [passwordConfirmation, setPasswordConfirmation] = useState('');
+  const [isSignup, setIsSignup] = useState(false);
+  const dispatch = useDispatch();
+  const { token, error, loading } = useSelector(state => state.auth);
+
+  function authHandler() {
+    if (isSignup) {
+      dispatch({ type: REGISTER_REQUEST, payload: { email, password, 'password_confirmation': passwordConfirmation } });
+    } else {
+      dispatch({ type: LOGIN_REQUEST, payload: { email, password } });
+      console.log('is async?');
+    }
+  }
+
+  useEffect(() => {
+    if (token) {
+      props.navigation.navigate({ routeName: 'Home' });
+    }
+  });
```

Comment:
Esto se va a correr en cada render donde haya un token. Si lo que quieren es ejecutarlo solo cuando el token cambio (supongo que eso sería lo correcto), tienen que agregar el token a la array de dependencias:
```suggestion
  useEffect(() => {
    if (token) {
      props.navigation.navigate({ routeName: 'Home' });
    }
  },
  [token]);
```

Reply:
Les recomiendo aplicar esta recomendación localmente en VSCode, no a través de github.